source/Irrlicht/CGUIEditBox.cpp

```diff
@@ -421,6 +421,19 @@ bool CGUIEditBox::processKey(const SEvent& event)
 	// Some special keys - but only handle them if KeyInput.Char is null as on some systems (X11) they might have same key-code as ansi-keys otherwise
 	else if (event.KeyInput.Char == 0)
 	{
+		// This may be a regular text key with Char set to 0 to explicitly ignore it.
+		// Absorb these so that they are not passed up to the parent.
+		// This is needed by the SDL driver, because SDL generates both a KEYDOWN and
+		// TEXTINPUT event for the same key press. To avoid duplicating, KEYDOWN
+		// events will almost always have Char == 0.
+		if (    (event.KeyInput.Key == KEY_SPACE) ||
+			(event.KeyInput.Key >= KEY_KEY_0 && event.KeyInput.Key <= KEY_KEY_Z) ||
+			(event.KeyInput.Key >= KEY_NUMPAD0 && event.KeyInput.Key <= KEY_DIVIDE) ||
+			(event.KeyInput.Key >= KEY_OEM_1 && event.KeyInput.Key <= KEY_OEM_102))
+		{
+			return true;
+		}

```

Comment:
Sorry but this is just piling more hacks ontop of hacks, why can't `CIrrDeviceSDL.cpp` do this correctly on its own?

Reply:
Fixed after the force push.